prototypes/javaspec-jupiter/src/test/java/info/javaspec/jupiter/syntax/subject/JavaSpec.java

```diff
@@ -0,0 +1,125 @@
+package info.javaspec.jupiter.syntax.subject;
+
+import org.junit.jupiter.api.DynamicContainer;
+import org.junit.jupiter.api.DynamicNode;
+import org.junit.jupiter.api.DynamicTest;
+import org.junit.jupiter.api.function.Executable;
+
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+import java.util.LinkedList;
+import java.util.Stack;
+import java.util.function.Supplier;
+
+final class JavaSpec<S> {
+  private final Stack<DynamicNodeList> containers = new Stack<>();
+  private Supplier<S> subjectSupplier;
+
+  public JavaSpec() {
+    //Push a null object onto the bottom of the stack, so there's always a parent list to add nodes to.
+    //Unlike all other entries on the stack, the root node list does not get turned into a DynamicContainer.
+    containers.push(new RootNodeList());
+  }
+
+  public void context(String condition, ContextBlock block) {
+    declareContainer(condition, block);
+  }
+
+  public DynamicNode describe(Class<?> actor, DescribeBlock block) {
+    return declareContainer(actor.getSimpleName(), block);
+  }
+
+  public DynamicNode describe(String functionOrGenericActor, DescribeBlock block) {
+    return declareContainer(functionOrGenericActor, block);
+  }
+
+  public DynamicNode disable(String intendedBehavior, Executable brokenVerification) {
+    DynamicTest test = makeSkippedTest(
+      intendedBehavior,
+      String.format("Disabled: %s.  This is not a failed assumption in the spec; it's just how JavaSpec disables a spec.", intendedBehavior)
+    );
+
+    addToCurrentContainer(test);
+    return test;
+  }
+
+  public DynamicTest it(String behavior, Executable verification) {
+    DynamicTest test = DynamicTest.dynamicTest(behavior, verification);
+    addToCurrentContainer(test);
+    return test;
+  }
+
+  public DynamicTest it(String behavior, ExecutableWithSubject<S> verification) {
+    DynamicTest test = DynamicTest.dynamicTest(behavior, () -> verification.execute(subject()));
```

Comment:
You forgot to use the new `it` form that passes the subject, when you tried Minimax.

Reply:
Updated to show how `it(String, (S subject) -> ...)` can be used.  It does the same thing, but it's a little more concise than calling `javaspec.subject()` all the time.